grammarinator/generate.py

```diff
@@ -286,11 +300,16 @@ def restricted_float(value):
                    transformers=args.transformer, serializer=args.serializer,
                    cleanup=False, encoding=args.encoding) as generator:
         if args.jobs > 1:
-            with Pool(args.jobs) as pool:
-                pool.map(generator, range(args.n))
+            with Manager() if not args.out else nullcontext() as manager:
+                if not args.out:
+                    generator = partial(generator, lock=manager.Lock())
+                with Pool(args.jobs) as pool:
+                    for _ in pool.imap_unordered(generator, count(0) if args.n == inf else range(args.n)):
+                        pass
         else:
-            for i in range(args.n):
-                generator(i)
+            lock = Manager().Lock() # Build an uncontested lock
```

Comment:
@renatahodovan We build a lock for the single-threaded case so that the [with lock](https://github.com/renatahodovan/grammarinator/pull/32/files#diff-2f19df232ab8dd27c19e9e80fe050965b0a7a7a43ed0503d06ce176486fa715dR145) call doesn't trip on `None`.

I think this is preferable to having null handling at that point, but let me know if you'd prefer a different approach!